pkg/oldhubble/cilium/dns.go

```diff
@@ -35,15 +34,20 @@ type FqdnCache interface {
 	AddDNSLookup(epID uint64, lookupTime time.Time, domainName string, ips []net.IP, ttl uint32)
 }
 
-// syncFQDNCache regularily syncs DNS lookups from Cilium into our local FQDN
+// syncFQDNCache regularly syncs DNS lookups from Cilium into our local FQDN
 // cache
 func (s *State) syncFQDNCache() {
+	t0 := 1 * time.Second
+	t := t0
 	for {
 		entries, err := s.ciliumClient.GetFqdnCache()
 		if err != nil {
 			s.log.WithError(err).Error("Unable to obtain fqdn cache from cilium")
-			time.Sleep(fqdnCacheRetryInterval)
+			time.Sleep(t)
+			t = 2 * t
```

Comment:
I know you tested it, but are we sure that the `s or s.ciliumClient ` are not nil?  if so fine, otherwise at least a nil check against s.ciliumClient could be helpful in future...

Also yes this is way better than messing up with state as the time is 64bits + we sleep so safe from any wrapping...

Reply:
AFAICT it's always not nil in our current code. Please let me know if you think otherwise. In any case, the nill check seems unrelated to this PR.

Reply:
Alright